pkg/deploy/generator/resources_rp.go

```diff
@@ -456,8 +468,19 @@ func (g *generator) rpVMSS() *arm.Resource {
 			},
 			Tags: map[string]*string{},
 			VirtualMachineScaleSetProperties: &mgmtcompute.VirtualMachineScaleSetProperties{
+				// Reference: https://learn.microsoft.com/en-us/azure/virtual-machine-scale-sets/virtual-machine-scale-sets-automatic-upgrade#arm-templates
 				UpgradePolicy: &mgmtcompute.UpgradePolicy{
-					Mode: mgmtcompute.UpgradeModeRolling,
+					Mode: mgmtcompute.UpgradeModeAutomatic,
+					RollingUpgradePolicy: &mgmtcompute.RollingUpgradePolicy{
+						// Percentage equates to 1.02 instances out of 3
```

Comment:
I don't think we should tolerate 2/3 instances running without alerting.  
This is to broad, and could put us in a worse position by not catching problems sooner.  
I created a story for creating an ICM automation workflow that will handle ICMs generated by the OS updates, I'm sure there has to be a log indicator for this, or status.
[ARO-9654](https://issues.redhat.com/browse/ARO-9654)

Reply:
I was thinking we should be able to tolerate 2/3 instances for a short period of time (e.g. 2 minutes, however long it takes for the rolling upgrade to complete), any longer and we get an alert. IMO, if we really need 3 instances running 100% of the time, we should be deploying more than 3 instances to account for VM uptime, so that our manual action in response for alerts for degraded instances is focused on preventing service degradation rather than mitigating it.

Reply:
Do we have understanding of when automatic updates are ran or is it just as-needed, outside of our control? Do we know how long they take to execute per-VM? It seems like we'd need that to make adjustments to our alerting.

I am in agreement that we should tolerate 1 unhealthy at a time, maximum, but only for a short period of time.

Reply:
It appears that this UpgradePolicy has a MaxSurge property in preview, which would handle scaling up new instances that are upgraded, and then rolling traffic over to the new instances before scaling down the old ones: https://learn.microsoft.com/en-us/azure/virtual-machine-scale-sets/virtual-machine-scale-sets-configure-rolling-upgrades. This might be something we consider in the future for this.

Reply:
This is being addressed in https://issues.redhat.com/browse/ARO-9889